src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1392,20 +1392,21 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
         }
         case SMART_ACTION_SET_DATA:
         {
+             // Areatrigger cannot be used as an invoker
             for (WorldObject* target : targets)
             {
                 if (Creature* cTarget = target->ToCreature())
                 {
                     CreatureAI* ai = cTarget->AI();
-                    if (IsSmart(cTarget))
+                    if (e.GetScriptType() != SMART_SCRIPT_TYPE_AREATRIGGER && IsSmart(cTarget))
```

Comment:
useless. IsSmart() returns false for ATs (Hence the error message). Invoker is never set so technically the action of setting invoker is set as it just goes to the else

Reply:
Which is also why you saw the result of "the action didn't get skipped"